### PR TITLE
implement adding tags + format all-notes output

### PR DESCRIPTION
**Pull request by UraFIZ**
The discussion over the implamentation of the adding a note you can find [here](https://mcscybersecurity2.slack.com/archives/C0738RX9XFZ/p1716416800314279)

Here is how the output of the all notes looks like:
image
It looks like UI cards with margin between them.

### CODE REVIEW

File: src/notes_handlers.py [hunk 1]
```diff
@@ -1,14 +1,18 @@
 from src.constants import INFO, UPDATED
 from src.models.Note import Note
 from src.models.NoteBook import NoteBook
-from src.decorators import input_error
+from src.decorators import collect_note_details, input_error
```
Comment (nrkshaman) [inline src/notes_handlers.py:4]: collect_note_details does not exist

File: src/notes_handlers.py [hunk 2]
```diff
@@ -49,6 +53,10 @@ def all_notes(notes:NoteBook) -> str:
...
-    for note in notes.values():
-        res += str(note)+"\n"+ "-"*30 + "\n"
-    return res.strip()
+    for title, note in notes.items():
+        res += "---------------------------------------------------------------------------------------------\n"
+        res += f"title:  |  {title}\n"
+        res += f"body:   |  {note.text}\n"
+        res += f"tags:   |  {', '.join(note.tags)}\n"
+        res += "---------------------------------------------------------------------------------------------\n"
+    return res
```
Comment (nrkshaman) [inline src/notes_handlers.py:62]: image
The long body is not supported

File: src/notes_handlers.py [hunk 3]
```diff
@@ -49,6 +53,10 @@ def all_notes(notes:NoteBook) -> str:
     if not notes.data:
         return INFO + " You do not have any notes saved"
     res=""
-    for note in notes.values():
-        res += str(note)+"\n"+ "-"*30 + "\n"
-    return res.strip()
+    for title, note in notes.items():
+        res += "---------------------------------------------------------------------------------------------\n"
```
Comment (nrkshaman) [inline src/notes_handlers.py:57]: and if we are here anyway, would be better to change this to "-"*80 (or whatever number)

**Review by nrkshaman — CHANGES_REQUESTED** [review]

File: src/notes_handlers.py (hunk 2 above)
Comment (UraFIZ) [reply src/notes_handlers.py:62]: let it be 100

Comment (UraFIZ) [reply src/notes_handlers.py:62]: Do you agree with this format of output all notes? If so, I am going to keep your advice and make changes in my branch. Could you please similary update the output of find-note?

File: src/notes_handlers.py (hunk 1 above)
Comment (UraFIZ) [reply src/notes_handlers.py:4]: yeah, i have removed it. Instead we go with "parse_input_add_note"

File: src/notes_handlers.py (hunk 2 above)
Comment (nrkshaman) [reply src/notes_handlers.py:62]: In general format is very nice. Regarding long body i would use `from textwrap import wrap` to allow multiline body with width 50-70
Also, could you move this formatting to Note __str__, so I could use it in find-note

File: src/notes_handlers.py (hunk 3 above)
Comment (UraFIZ) [reply src/notes_handlers.py]: I wrote that i had surprised coz I could see your review online in my VSC. It's was for the first time:)

Comment (UraFIZ) [reply src/notes_handlers.py]: Thank you. Done.

File: src/notes_handlers.py (hunk 1 above)
Comment (UraFIZ) [reply src/notes_handlers.py]: Done.

**Review by nrkshaman — APPROVED** [review]

**Pull request merged**